src/components/frame/Extensions/pages/ExtensionDetailsPage/ExtensionDetailsPage.tsx

```diff
@@ -82,50 +102,71 @@ const ExtensionDetailsPage: FC = () => {
     navigate,
     refetchColony,
     refetchExtensionData,
-    setWaitingForEnableConfirmation,
+    setWaitingForActionConfirmation,
   });
 
   const SetupComponent = SetupComponentMap[extensionData.extensionId];
 
+  const actionType =
+    isStakedExpenditureExtension &&
+    extensionData.isInitialized &&
+    extensionData.isEnabled
+      ? ActionTypes.SET_STAKE_FRACTION
+      : ActionTypes.EXTENSION_ENABLE;
```

Comment:
There will be other extensions with a settings tab like this, so can this be made more generic to support other action types depending on the extension rather than a ternary?